packages/api/src/task/scheduler.ts

```diff
@@ -264,6 +277,36 @@ export class TaskScheduler {
     await this.updateTask(task, { status });
   }
 
+  async retryTask(task: WithID<Task>, event?: messages.TaskResult) {
+    let attempts = task.status.attempts + 1 ?? 1;
+    if (attempts <= MAX_RETRY_ATTEMPTS) {
+      const status: Task["status"] = {
+        ...task.status,
+        attempts: attempts,
+      };
+
+      await this.updateTask(task, { status });
+      task.status = status;
```

Comment:
You should not modify objects in-place here, can lead to several bugs (e.g. maybe `updateTask` is asynchronously sending the rabbitmq event for the task.udpated webhook, then you're messing with the object in the middle of the serialization here).

So you should always give preference to creating new objects with updated fields (using spread syntax `task = { ...task, status }`) instead, or better yet, WDYT of having `updateTask` return the updated task? This will be especially useful if `updateTask` starts doing the merging of the `status` instead of just setting whatever the client sends (I suggested that in the other comment you asked about the repeated `...task.status`).

Then this is just
```suggestion
      task = await this.updateTask(task, { status });
```

Reply:
(this whole scheduler logic always reminds me that we need some SQL transactions here 🥲 )